ztag/transforms/ssh.py

```diff
@@ -3,58 +3,151 @@
+        set_value(support, 'server_to_client', server_to_client)
+
+        set_value(out, 'support', support)
+
+        # selected:
+        selected = grab['algorithm_selection'].resolve()
+        if selected is not None:
+            rename_key(selected, 'dh_kex_algorithm', 'kex_algorithm')
+            rename_key(selected, 'client_to_server_alg_group', 'client_to_server')
+            rename_key(selected, 'server_to_client_alg_group', 'server_to_client')
+        set_value(out, 'selected', selected)
+
+        # key_exchange:
+        key_exchange = dict()
+        set_value(key_exchange, 'ecdh_params', grab['key_exchange']['ecdh_params'].resolve())
+        set_value(key_exchange, 'dh_params', grab['key_exchange']['dh_params'].resolve())
+        try:
+            del key_exchange['dh_params']['server_public']
+        except KeyError:
+            pass
+        set_value(out, 'key_exchange', key_exchange)
+
+        # server_host_key:
+        host_key = dict()
+        for clone_key in ['fingerprint_sha256', 'rsa_public_key', 'dsa_public_key', 'ecdsa_public_key', 'ed25519_public_key']:
+            # a bunch of keys we just need to copy verbatim from the grab
+            set_value(host_key, clone_key, grab['key_exchange']['server_host_key'][clone_key].resolve())
+        set_value(host_key, 'key_algorithm', grab['key_exchange']['server_host_key']['algorithm'].resolve())
+
+        certkey_public_key = grab['key_exchange']['server_host_key']['certkey_public_key'].resolve()
+        if certkey_public_key is not None:
+            rename_key(certkey_public_key, 'cert_type', 'type')
+            del_key(certkey_public_key, 'reserved')
+
+            signature_key = certkey_public_key.get('signature_key')
+            if signature_key is not None:
+                del_key(signature_key, 'raw')
+                rename_key(signature_key, 'algorithm', 'key_algorithm')
+                certkey_public_key['signature_key'] = signature_key
+
+            # rewrite extensions and critical_options to maps of string -> bool:
+            extensions = certkey_public_key.get('extensions')
+            if extensions is not None:
+                certkey_public_key['extensions'] = rewrite_known(extensions)
+            critical_options = certkey_public_key.get('critical_options')
+            if critical_options is not None:
+                certkey_public_key['critical_options'] = rewrite_known(critical_options)
+
+            key = certkey_public_key.get('key')
+            if key is not None:
+                del_key(key, 'raw')
+                certkey_public_key['key'] = key
+
+            sig = certkey_public_key.get('signature')
+            if sig is not None:
+                formatted_sig = dict()
+                parsed = sig.get('parsed')
+                if parsed is not None:
+                    set_value(formatted_sig, 'value', parsed.get('value'))
```

Comment:
I'm usually explicit about defaults of `None` when calling `.get()`, but I just learned it didn't throw in this case, so I'm probably wrong.

TIL.